api/client.go

```diff
@@ -77,6 +77,7 @@ func (cfg *Config) validate() error {
 type Client interface {
 	URL(ep string, args map[string]string) *url.URL
 	Do(context.Context, *http.Request) (*http.Response, []byte, error)
+	CloseIdleConnections()
```

Comment:
I double check and I would propose we remove `CloseIdleConnections()` from both interfaces. We don't want to break compatibility for not too important reason (we kind of can because it's an experimental package, but only if we have to).

This method on interface is not needed because you can totally do 
```

type closeIdler interface {
	CloseIdleConnections()
}

// .. in some function
cl, err := NewClient()
// check err
cl.(closeIdler).CloseIdleConnection()
```

Thus to be safe, I would start with this approach first. It unblocks you without impacting potentially many packages in a very painful way (when upgrading client_golang).